### PR TITLE
feat: ether to smart wallet payment

### PR DESCRIPTION
**Pull request by leoslr**
## Description of the changes

Enables to pay requests in ether when the recipient is a proxy smart contract, like smart wallet such as Gnosis Safe.

The `.transfer` is limited to 2300 gas, so when a payment recipient was a Gnosis Safe, we did not pass enough gas for it to execute it's fallback function. I changed it to `.call` which pass all the remaining transaction gas.

New tests cover for this specific case.

### CODE REVIEW

File: packages/smart-contracts/src/contracts/test/EtherFallbackPayment.sol [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+contract EtherPaymentFallback {
+    event SafeReceived(address indexed sender, uint256 value);
+
+    /// @dev Fallback function accepts Ether transactions.
+    receive() external payable {
+        emit SafeReceived(msg.sender, msg.value);
+    }
+
+    constructor() {}
+}
```
Comment (leoslr) [inline packages/smart-contracts/src/contracts/test/EtherFallbackPayment.sol:15]: Gnosis Safe extends this contract to allow for ether payments, no need to import the whole Gnosis Safe Contract.
(Tests fail with our previous EthFeeProxy and EthProxy).

File: packages/smart-contracts/src/contracts/EthereumFeeProxy.sol [hunk 2]
```diff
@@ -1,78 +1,75 @@
...
-    nonReentrant
-    public
-    payable
-  {
-    _to.transfer(_amount);
-    _feeAddress.transfer(_feeAmount);
-    // transfer the remaining ethers to the sender
-    payable(msg.sender).transfer(msg.value - _amount - _feeAmount);
+        // transfer the remaining ethers to the sender
+        (bool sendBackSuccess, ) = payable(msg.sender).call{
```
Comment (leoslr) [inline packages/smart-contracts/src/contracts/EthereumFeeProxy.sol:68]: Modified `.transfer` to `.call` here too. This way we cover the case where a gnosis safe is the one initiating a payment and we have to send ether back to it.

**Review by bertux — APPROVED** [review]
This seems a  good extension :-)

**Review by benjlevesque — APPROVED** [review]
very thourough, nice work

**Pull request merged**